### PR TITLE
Support more granular AgentStates instead of ERROR

### LINKED ISSUE #17: Agent: support new AgentStates
Instead of current `ERROR` state the following states need to be added and supported:
* `CLI_FAILED` when SAVE CLI returns non-zero status
* `BACKEND_FAILURE` when backend returns non-OK code while uploading test results
* `BACKEND_UNREACHABLE` when backend doesn't respond

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Added states CLI_FAILED, BACKEND_FAILED, BACKEND_UNREACHABLE
* Changed logic in Agent

Closes #17

### CODE REVIEW

File: save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt [hunk 1]
```diff
@@ -128,18 +139,25 @@ class SaveAgent(private val config: AgentConfiguration,
             val result = runCatching {
                 postExecutionData(executionData)
             }
-            if (result.isSuccess) {
+            if (result.isSuccess && result.getOrNull()?.statusCode == HttpStatusCode.OK) {
                 return@repeat
```
Comment (aktsay6) [inline save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt:143]: Why do we return at repeat here?

Comment (petertrr) [reply save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt:143]: I guess I initially planned to add some logic after `repeat`, but now it has no difference from usual `return`. I'll change.

Comment (petertrr) [reply save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt:140]: Actually, `repeat` is a non-inlined function and therefore plain `return` is forbidden there. So I added it back :)

**Review by aktsay6 — APPROVED** [review]
lgtm

**Pull request merged**